### ISSUE TITLE
[Feature Request] Show notes when viewing a task

### ISSUE THREAD

**Issue by JPTIZ** [lfos/calcurse#289]
Started using calcurse today and it is actually pretty nice and convenient :) But speaking of convenience...

An opened task window is almost the size of the whole application, but it only shows the title (which is expected to be oneliner - at least in my understanding), while for notes there is a separate command to show notes (but it doesn't show the task's title).

I think it would make more sense to actually show the notes in that big task window, so I can quickly check the notes without having to type `>`. The `>` could be left to use when notes become too big to fit the window.

**Comment by Adam-Antios** [comment]
Yes! Exactly that! It is peculiar for this to not be the normal behaviour!

**Comment by rosalogia** [comment]
I tried implementing this and ran into some minor problems, and furthermore am unsure as to whether I am doing this responsibly since I don't really write C.

You can see which parts are my implementation from the little green `+`s indicating additions on my end:
![todonoteimpl](https://user-images.githubusercontent.com/66163965/94519236-41a85480-01f8-11eb-8e48-acd91ed05e9c.png)

Here's how it looks in use:
![notepic](https://user-images.githubusercontent.com/66163965/94519454-816f3c00-01f8-11eb-94ff-1a75aa2427ec.png)

However, I'm finding that closing the TODO view (with `v`) and opening it again in place (with `v` again) causes this little rendering bug (see the little `^?` at the bottom?):
![notebug](https://user-images.githubusercontent.com/66163965/94519536-b11e4400-01f8-11eb-905e-a8d8f77da8db.png)

I have no clue what's causing it, so any pointers on where to start looking for the issue would be appreciated.

**Comment by JPTIZ** [comment]
Wow, nice job, @rosalogia! :) 

At first glance it __looks__ like some null-terminated string issue, although you do seem to take care of it. Can you double-check it? If it is the case, I think it would happen in that `msg` variable.

Also, what's the branch you're working on so I can try something here?

**Comment by rosalogia** [comment]
Thank you!

I will happily double check, but you can take a look on [my fork](https://github.com/rosalogia/calcurse/commit/aa811d9dd5495031c95af98822f160767bc18386).

**Comment by rosalogia** [comment]
Problem resolved! It was a null-terminator issue, and that's been [fixed](https://github.com/rosalogia/calcurse/commit/91272069cb894278fd241aac3e8458280bdc35da). It's working pretty well at this point, I don't know if anything else should be done before a PR is considered. It could definitely use some review and further testing, so that would be much appreciated from anyone with the time.

**Comment by JPTIZ** [comment]
For me it looks like a ready-to-review patch :) Although this repo [do have a contributing guideline](https://github.com/rosalogia/calcurse/blob/pu/doc/submitting-patches.txt), it seems to be really outdated (file added in 2013 and since then there were only license updates), since PRs seem to be accepted in this repo straight forwardly.

About review, you can let it be done in the PR itself. I would in advance just tell about spacing patterns (see [this misindent here in line 325?](https://github.com/lfos/calcurse/compare/pu...rosalogia:notes-in-todo-view#diff-f66e70a66349c055fdcd4704456dfc7cR325)). If the file uses tabs, go all tabs on that one. If it uses spaces, go all spaces. If it mixes...well...we need an additional formatting PR then haha.

(Slightly off-topic: this seems to be a nice candidate PR for hacktoberfest, if you want)

**Comment by rosalogia** [comment]
Sounds good! Thanks for your help :) I'll probably wait until October to make the PR, but till then I'll look around and see what else I can help with 👍

**Comment by rosalogia** [comment]
Alright, I believe the formatting issues were due to me using spaces while the rest of the codebase was using tabs, so that's been resolved. I was also able to make this behaviour apply to events and appointments. However, I found myself repeating code a lot since I'm not sure how to safely abstract away the common parts of the note-reading process in C. Hopefully this is something we can address once a PR is up!
